File: 2OLIDTS-WilliamSelvas-05IA/2OLIDTS-WilliamSelvas-05IA.cpp
// 2OLIDTS-WilliamSelvas-05IA.cpp : Programa de ejemplo para manejar un vector de nombres de aves.
// Contiene funciones para mostrar, agregar y eliminar aves usando std::vector.

#include <iostream>
#include <vector>
#include <string>

using namespace std;

// Muestra las aves que están en el vector
void arreglo_aves(const vector<string>& aves)
{
    if (aves.empty()) {
        cout << "No hay aves en el vector." << '\n';
        return;
    }

    cout << "Los valores del vector:" << '\n';
    for (size_t i = 0; i < aves.size(); ++i) {
        cout << aves[i] << " ";
    }
    cout << "\n";
}

// Agrega tres aves al final del vector usando push_back (modifica el vector por referencia)
void push_arreglo_aves(vector<string>& aves)
{
    aves.push_back("Mayna");
    aves.push_back("Periquitos");
    aves.push_back("Cacatua");

    cout << "Los valores del vector despues de insertar: \n";
    for (size_t i = 0; i < aves.size(); ++i) {
        cout << aves[i] << " ";
    }
    cout << "\n";
}

// Elimina la última ave del vector usando pop_back (modifica el vector por referencia)
void pop_arreglo_aves(vector<string>& aves)
{
    if (aves.empty()) {
        cout << "No hay aves para eliminar." << '\n';
        return;
    }
    aves.pop_back();
    cout << "Los valores del vector despues del pop:\n";
    for (size_t i = 0; i < aves.size(); ++i) {
        cout << aves[i] << " ";
    }
    cout << "\n";
}

int main()
{
    vector<string> aves = { "Loro gris", "Paloma de diamante", "Coctel" };

    arreglo_aves(aves);
    push_arreglo_aves(aves);
    pop_arreglo_aves(aves);

    return 0;
}

File: 2OLIDTS-WilliamSelvas-05IA/2OLIDTS-WilliamSelvas-05IA.py
# aves.py - Programa que maneja una lista de nombres de aves.

def mostrar_aves(aves):
    if not aves:
        print("No hay aves en la lista.")
        return
    print("Los valores de la lista:")
    print(" ".join(aves))
    print()

def push_arreglo_aves(aves):
    # Agrega tres aves al final (equivalente a push_back)
    aves.append("Mayna")
    aves.append("Periquitos")
    aves.append("Cacatua")
    print("Los valores de la lista despues de insertar:")
    print(" ".join(aves))
    print()

def pop_arreglo_aves(aves):
    # Elimina la última ave (equivalente a pop_back)
    if not aves:
        print("No hay aves para eliminar.")
        return
    eliminado = aves.pop()
    print(f"Eliminando la última ave: {eliminado}")
    print("Los valores de la lista despues del pop:")
    print(" ".join(aves))
    print()


def main():
    aves = ["Loro gris", "Paloma de diamante", "Coctel"]

    mostrar_aves(aves)
    push_arreglo_aves(aves)
    pop_arreglo_aves(aves)


if __name__ == '__main__':
    main()
